tools/ci/checks/get_changeset.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# Get the set of files changed relative to a specifed git reference.
+# The COMPARE_TO environment variable is used, and if no value is specified
```

Comment:
Well, the logic is actually slightly different: if HEAD is invalid, then $COMPARE_TO will be ignored, no?